### PR TITLE
drivers: flash: spi_flash_at45: Use gpio_pin_set_dt helper

### PR DESCRIPTION
**Pull request by pdgendt**
Replace occurrences of `gpio_pin_set` with the `_dt` helper.

### CODE REVIEW

**Review by galak — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

File: drivers/flash/spi_flash_at45.c [hunk 1]
```diff
@@ -549,23 +549,20 @@ static int spi_flash_at45_init(const struct device *dev)
 
 #if ANY_INST_HAS_RESET_GPIOS
 	if (dev_config->reset) {
-		if (gpio_pin_configure_dt(dev_config->reset,
-					GPIO_OUTPUT_ACTIVE)) {
+		if (gpio_pin_configure_dt(dev_config->reset, GPIO_OUTPUT_ACTIVE)) {
```
Comment (henrikbrixandersen) [inline drivers/flash/spi_flash_at45.c:552]: This lacks a call to `device_is_ready()`.

File: drivers/flash/spi_flash_at45.c [hunk 2]
```diff
@@ -549,23 +549,20 @@ static int spi_flash_at45_init(const struct device *dev)
...
-		gpio_pin_set(dev_config->reset->port, dev_config->reset->pin, 0);
+		gpio_pin_set_dt(dev_config->reset, 0);
 	}
 #endif
 
 #if ANY_INST_HAS_WP_GPIOS
 	if (dev_config->wp) {
-		if (gpio_pin_configure_dt(dev_config->wp,
-					GPIO_OUTPUT_ACTIVE)) {
+		if (gpio_pin_configure_dt(dev_config->wp, GPIO_OUTPUT_ACTIVE)) {
```
Comment (henrikbrixandersen) [inline drivers/flash/spi_flash_at45.c:562]: This lacks a call to `device_is_ready()`.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

File: drivers/flash/spi_flash_at45.c [hunk 3]
```diff
@@ -549,23 +549,28 @@ static int spi_flash_at45_init(const struct device *dev)
...
+		if (!device_is_ready(dev_config->reset->port)) {
+			LOG_ERR("Reset pin not ready");
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(dev_config->reset, GPIO_OUTPUT_ACTIVE)) {
 			LOG_ERR("Couldn't configure reset pin");
 			return -ENODEV;
 		}
-		gpio_pin_set(dev_config->reset->port, dev_config->reset->pin, 0);
+		gpio_pin_set_dt(dev_config->reset, 0);
```
Comment (nordicjm) [inline drivers/flash/spi_flash_at45.c:560]: No point setting it up as active to them make it inactive, can just use `GPIO_OUTPUT_INACTIVE` instead

Comment (pdgendt) [reply drivers/flash/spi_flash_at45.c:560]: > No point setting it up as active to them make it inactive, can just use `GPIO_OUTPUT_INACTIVE` instead

It has an internal reset controller, we need to assert to trigger.

**Review by nordicjm — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**